### PR TITLE
Fix #2937 - add Encoding param codepage support

### LINKED ISSUE #2937: Encoding parameter accept numeric Ids and name of registered code pages
Issue Details
-----------------

PR https://github.com/PowerShell/PowerShell/pull/7636  enhance the encoding transformation attribute:
- to allow numeric Ids of registered code pages (like -Encoding 1251)
- to allow string names of registered code pages (like -Encoding "windows-1251")

Affected cmdlets:
```powershell
gcm -ParameterName Encoding

CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
Cmdlet          Add-Content                                        6.1.0.0    Microsoft.PowerShell.Management
Cmdlet          Export-Clixml                                      6.1.0.0    Microsoft.PowerShell.Utility
Cmdlet          Export-Csv                                         6.1.0.0    Microsoft.PowerShell.Utility
Cmdlet          Export-PSSession                                   6.1.0.0    Microsoft.PowerShell.Utility
Cmdlet          Format-Hex                                         6.1.0.0    Microsoft.PowerShell.Utility
Cmdlet          Get-Content                                        6.1.0.0    Microsoft.PowerShell.Management
Cmdlet          Import-Csv                                         6.1.0.0    Microsoft.PowerShell.Utility
Cmdlet          Out-File                                           6.1.0.0    Microsoft.PowerShell.Utility
Cmdlet          Select-String                                      6.1.0.0    Microsoft.PowerShell.Utility
Cmdlet          Send-MailMessage                                   6.1.0.0    Microsoft.PowerShell.Utility
Cmdlet          Set-Content                                        6.1.0.0    Microsoft.PowerShell.Management
```

Full list supported code pages we can found in
https://github.com/dotnet/corefx/blob/master/src/System.Text.Encoding.CodePages/src/System/Text/EncodingTable.Data.cs

Version(s) of document impacted
------------------------------
- [x] Impacts 6.next document
- [ ] Impacts 6 document
- [ ] Impacts 5.1 document
- [ ] Impacts 5.0 document
- [ ] Impacts 4.0 document
- [ ] Impacts 3.0 document

Reason(s) for not selecting all version of documents
----------------------------------------------------
- [x] The documented feature was introduced in selected version of PowerShell
- [ ] This issue only shows up in selected version of the document

### PR DESCRIPTION
**Pull request by sdwheeler**
Version(s) of document impacted
------------------------------
- [ ] Impacts 6.next document
- [x] Impacts 6 document
- [ ] Impacts 5.1 document
- [ ] Impacts 5.0 document
- [ ] Impacts 4.0 document
- [ ] Impacts 3.0 document

Reason(s) for not updating all version of documents
--------------------------------------------------
- [ ] The documented feature was introduced in version (list version here) of PowerShell
- [ ] This issue only shows up in version (list version(s) here) of the document
- [ ] This PR partially fixes the issue, and issue #<insert here> tracks the remaining work

### CODE REVIEW

**Comment by zjalexander** [comment]
Docs Build status updates of commit _[0c0a66a](https://github.com/sdwheeler/PowerShell-Docs/pull/4178/commits/0c0a66a785c38fc77a991a894dc0ae56063bb2b7)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[reference/6/Microsoft.PowerShell.Management/Add-Content.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue2937/reference/6/Microsoft.PowerShell.Management/Add-Content.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.management/add-content?view=powershell-6&branch=pr-en-us-4178) (powershell-6) |
[reference/6/Microsoft.PowerShell.Management/Get-Content.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue2937/reference/6/Microsoft.PowerShell.Management/Get-Content.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.management/get-content?view=powershell-6&branch=pr-en-us-4178) (powershell-6) |
[reference/6/Microsoft.PowerShell.Management/Set-Content.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue2937/reference/6/Microsoft.PowerShell.Management/Set-Content.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.management/set-content?view=powershell-6&branch=pr-en-us-4178) (powershell-6) |
[reference/6/Microsoft.PowerShell.Utility/Export-Clixml.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue2937/reference/6/Microsoft.PowerShell.Utility/Export-Clixml.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/export-clixml?view=powershell-6&branch=pr-en-us-4178) (powershell-6) |
[reference/6/Microsoft.PowerShell.Utility/Export-Csv.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue2937/reference/6/Microsoft.PowerShell.Utility/Export-Csv.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/export-csv?view=powershell-6&branch=pr-en-us-4178) (powershell-6) |
[reference/6/Microsoft.PowerShell.Utility/Export-PSSession.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue2937/reference/6/Microsoft.PowerShell.Utility/Export-PSSession.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/export-pssession?view=powershell-6&branch=pr-en-us-4178) (powershell-6) |
[reference/6/Microsoft.PowerShell.Utility/Format-Hex.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue2937/reference/6/Microsoft.PowerShell.Utility/Format-Hex.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/format-hex?view=powershell-6&branch=pr-en-us-4178) (powershell-6) |
[reference/6/Microsoft.PowerShell.Utility/Import-Csv.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue2937/reference/6/Microsoft.PowerShell.Utility/Import-Csv.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/import-csv?view=powershell-6&branch=pr-en-us-4178) (powershell-6) |
[reference/6/Microsoft.PowerShell.Utility/Out-File.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue2937/reference/6/Microsoft.PowerShell.Utility/Out-File.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/out-file?view=powershell-6&branch=pr-en-us-4178) (powershell-6) |
[reference/6/Microsoft.PowerShell.Utility/Select-String.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue2937/reference/6/Microsoft.PowerShell.Utility/Select-String.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/select-string?view=powershell-6&branch=pr-en-us-4178) (powershell-6) |
[reference/6/Microsoft.PowerShell.Utility/Send-MailMessage.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue2937/reference/6/Microsoft.PowerShell.Utility/Send-MailMessage.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/send-mailmessage?view=powershell-6&branch=pr-en-us-4178) (powershell-6) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C4%5C19%5C6ff7e8c3-dfc6-3ebd-da5a-d5e2ff43de8f%5CPullRequest%5C201904192032507236-4178%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=IQJ2slA4fW2zKmPMo9H8GDgMlodEqnYGlHBnx8pmi%2Bw%3D&st=2019-04-19T20%3A32%3A02Z&se=2019-05-20T20%3A37%3A02Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: reference/6/Microsoft.PowerShell.Management/Add-Content.md [hunk 1]
```diff
@@ -221,6 +221,11 @@ The acceptable values for this parameter are as follows:
 - **UTF32**: Encodes in UTF-32 format.
 - **Unknown**: The encoding type is unknown or invalid; the data can be treated as binary.
 
+Beginning with PowerShell 6.2, the **Encoding** parameter also allows numeric IDs of registered code
+pages (like `-Encoding 1251`) or string names of registered code pages (like
+`-Encoding "windows-1251"`). For more information, see the .NET documentation for
+[Encoding.CodePage](https://docs.microsoft.com/en-us/dotnet/api/system.text.encoding.codepage?view=netcore-2.2).
```
Comment (DCtheGeek) [inline reference/6/Microsoft.PowerShell.Management/Add-Content.md]: Fix the link to remove FQDN and localization.  This looks true in all the files in this PR.

Comment (sdwheeler) [reply reference/6/Microsoft.PowerShell.Management/Add-Content.md]: fixed

**Comment by zjalexander** [comment]
Docs Build status updates of commit _[e2e9d98](https://github.com/sdwheeler/PowerShell-Docs/pull/4178/commits/e2e9d98677f770ad642674e8be335f23b8d9f0f1)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[reference/6/Microsoft.PowerShell.Management/Add-Content.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue2937/reference/6/Microsoft.PowerShell.Management/Add-Content.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.management/add-content?view=powershell-6&branch=pr-en-us-4178) (powershell-6) |
[reference/6/Microsoft.PowerShell.Management/Get-Content.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue2937/reference/6/Microsoft.PowerShell.Management/Get-Content.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.management/get-content?view=powershell-6&branch=pr-en-us-4178) (powershell-6) |
[reference/6/Microsoft.PowerShell.Management/Set-Content.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue2937/reference/6/Microsoft.PowerShell.Management/Set-Content.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.management/set-content?view=powershell-6&branch=pr-en-us-4178) (powershell-6) |
[reference/6/Microsoft.PowerShell.Utility/Export-Clixml.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue2937/reference/6/Microsoft.PowerShell.Utility/Export-Clixml.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/export-clixml?view=powershell-6&branch=pr-en-us-4178) (powershell-6) |
[reference/6/Microsoft.PowerShell.Utility/Export-Csv.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue2937/reference/6/Microsoft.PowerShell.Utility/Export-Csv.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/export-csv?view=powershell-6&branch=pr-en-us-4178) (powershell-6) |
[reference/6/Microsoft.PowerShell.Utility/Export-PSSession.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue2937/reference/6/Microsoft.PowerShell.Utility/Export-PSSession.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/export-pssession?view=powershell-6&branch=pr-en-us-4178) (powershell-6) |
[reference/6/Microsoft.PowerShell.Utility/Format-Hex.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue2937/reference/6/Microsoft.PowerShell.Utility/Format-Hex.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/format-hex?view=powershell-6&branch=pr-en-us-4178) (powershell-6) |
[reference/6/Microsoft.PowerShell.Utility/Import-Csv.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue2937/reference/6/Microsoft.PowerShell.Utility/Import-Csv.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/import-csv?view=powershell-6&branch=pr-en-us-4178) (powershell-6) |
[reference/6/Microsoft.PowerShell.Utility/Out-File.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue2937/reference/6/Microsoft.PowerShell.Utility/Out-File.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/out-file?view=powershell-6&branch=pr-en-us-4178) (powershell-6) |
[reference/6/Microsoft.PowerShell.Utility/Select-String.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue2937/reference/6/Microsoft.PowerShell.Utility/Select-String.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/select-string?view=powershell-6&branch=pr-en-us-4178) (powershell-6) |
[reference/6/Microsoft.PowerShell.Utility/Send-MailMessage.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue2937/reference/6/Microsoft.PowerShell.Utility/Send-MailMessage.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/send-mailmessage?view=powershell-6&branch=pr-en-us-4178) (powershell-6) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C4%5C20%5C6ff7e8c3-dfc6-3ebd-da5a-d5e2ff43de8f%5CPullRequest%5C201904200118394920-4178%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=slJbKDzjNYqkJzwdZzhe8ddJGRc9vhY6eFdOCIsWxAc%3D&st=2019-04-20T01%3A17%3A50Z&se=2019-05-21T01%3A22%3A50Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Pull request merged**